xla/service/gpu/nccl_collective_thunk.cc

```diff
@@ -224,6 +224,49 @@ StatusOr<std::vector<DeviceBufferPair>> ConvertToDeviceBuffers(
   return device_buffers;
 }
 
+Status MaybeRegisterBuffers(int device_ordinal,
+                            const std::vector<DeviceBufferPair>& buffers,
+                            ncclComm_t comm) {
+  // TODO(tmorris): Only register if xla_gpu_enable_nccl_user_buffers is true.
```

Comment:
If we don't, what happens?

Reply:
`ncclCommRegister` is a no-op if the buffer doesn't meet the requirements that `ncclMemAlloc` provides. However, I think I would still like to plumb the `xla_gpu_enable_nccl_user_buffers` flag through.

Reply:
I assume it's still safer to check if user buffer is used here.
Can you pass the bool flag from debug_options through RunAllReduce/RunAllGather/OtherOperation methods?
Here: https://github.com/openxla/xla/blob/main/xla/service/gpu/runtime/collectives.cc#L553
And Here: https://github.com/openxla/xla/blob/main/xla/service/gpu/runtime/collectives.cc#L478
Etc?

@cheshire WDYT?